File: app/src/main/java/com/yiche/example/newestversiondemo/adapter/LieAdapter.java
package com.yiche.example.newestversiondemo.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.yiche.example.newestversiondemo.R;

import java.util.ArrayList;
import java.util.List;

/**
 * <p>Created by shixin on 2018/3/6.
 */
public class LieAdapter extends RecyclerView.Adapter<LieAdapter.ViewHolder> {
    private List<String> countryList = new ArrayList<>();

    public void setList(List<String> list) {
        countryList.clear();
        if(list!=null&&!list.isEmpty()) {
            countryList.addAll(list);
        }
        notifyDataSetChanged();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.adapter_lie, parent, false));
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.tvTitle.setText(countryList.get(position));
    }

    @Override
    public int getItemCount() {
        return countryList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvTitle;
        public ViewHolder(View itemView) {
            super(itemView);
            tvTitle = itemView.findViewById(R.id.tv_title);
        }
    }
}

File: app/src/main/java/com/bride/demon/adapter/LieAdapter.kt
package com.bride.demon.adapter

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.bride.demon.R

import java.util.ArrayList

class LieAdapter : RecyclerView.Adapter<LieAdapter.ViewHolder>() {
    private val countryList = ArrayList<String>()

    fun setList(list: List<String>?) {
        countryList.clear()
        if (!list.isNullOrEmpty()) {
            countryList.addAll(list)
        }
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.adapter_lie, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tvTitle.text = countryList[position]
    }

    override fun getItemCount(): Int {
        return countryList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var tvTitle: TextView = itemView.findViewById(R.id.tv_title)
    }
}
